HW-13

JerryngtonD opened the pull request:
- written unit tests

JerryngtonD committed:
HW-13
- written unit tests

JerryngtonD committed:
HW-13
- written integration tests

JerryngtonD committed:
HW-13
- written ui-test

JerryngtonD committed:
HW-13
- fixed old staff

JerryngtonD committed:
HW-13
- rollback to old version espresso because bug with navigation drawer
- added check favorites list

JerryngtonD committed:
HW-13
- fixed favorites list viewing (error with incorrect url at glide)

JerryngtonD committed:
HW-13
- added custom toast matcher

JerryngtonD committed:
HW-13
- written test when internet is switched off (don't forget to switch off internet)

shustreek commented on app/src/test/java/ru/otus/cineman/integration/MovieInteractorIntegrationTest.kt:
```diff
@@ -0,0 +1,152 @@
+            }
+
+        movieService = retrofit.create(MovieService::class.java)
+        movieDb = Db.getInstance(InstrumentationRegistry.getInstrumentation().targetContext)
```
на будущее. тесты всегда запускаются с нуля. т.е. приложение не установлено и не разу не запускалось. на данном этапе у меня тест не проходит, т.к. бд не проинициализировалась. для тестов БД создавать в памяти 'Room.inMemoryDatabaseBuilder'

JerryngtonD replied:
заменил на inMemoryDb, действительно необходимая штука, жаль в контейнерах тесты для мобилки нельзя - не знаю как гонять.

shustreek commented on app/src/androidTest/java/ru/otus/cineman/AddToFavoritesUITest.kt:
```diff
@@ -0,0 +1,35 @@
+            )
+        )
+
+        onView(withId(R.id.drawer_layout)).perform(open());
```
тесты нужно писать для каждого отдельный метод
т.е. тут можно было сделать так метод для проверки кнопки, метод для проверки перехода в избранное. в этом случае можно было бы сделать после перехода проверку на то, что в списке он добавился.

JerryngtonD replied:
перепишу в дальнейшем )

shustreek commented on app/src/androidTest/java/ru/otus/cineman/GettingErrorWithoutInternetUITest.kt:
```diff
@@ -0,0 +1,28 @@
+    val activityRule = ActivityTestRule(MainActivity::class.java)
+
+    @Test
+    fun checkErrorCauseWithoutInternet() {
```
тест не проходит, даже если я вырубаю инет. при переходе на этот экран просто нет загрузки и ошибка не появляется. появляется только при рефреше

JerryngtonD replied:
добавил swipeToRefresh action до того момента,
как проверяю snackbar. Вроде теперь срабатывает.

JerryngtonD committed:
HW-13
- added inMemory db got test
- added swipe to refresh to get error cause internet

JerryngtonD committed:
HW-13
- added IdlingResource for async test
